### PR TITLE
Add comment when disabling autopush on negative karma

### PR DESCRIPTION
**Pull request by trishnaguha**
Fixes https://github.com/fedora-infra/bodhi/issues/1291

### CODE REVIEW

File: bodhi/server/scripts/approve_testing.py
Comment (bowlofeggs) [inline bodhi/server/scripts/approve_testing.py]: I think it would be desirable to make this comment at the moment that the negative karma is disabled, rather than later. This script only runs every 6 hours. I think this would be a good place to add such a comment:

https://github.com/fedora-infra/bodhi/blob/2.4.0/bodhi/server/models/models.py#L1724-L1726

**Review by bowlofeggs — CHANGES_REQUESTED** [review]

File: bodhi/server/scripts/approve_testing.py
Comment (trishnaguha) [reply bodhi/server/scripts/approve_testing.py]: Makes sense! I never knew it runs every 6 hours.

File: bodhi/tests/server/services/test_updates.py
Comment (trishnaguha) [inline bodhi/tests/server/services/test_updates.py]: @bowlofeggs Looks good :-) ?

**Comment by trishnaguha** [comment]
@bowlofeggs Updated :-).

File: bodhi/server/models.py
Comment (bowlofeggs) [inline bodhi/server/models.py]: It would be a good idea to put a default value here, so that Bodhi won't crash if this new config value isn't added to the config. I like to follow [Semantic Versioning](https://semver.org) where possible, so this would be an easy way to avoid a backwards incompatible change.

I think this might be as simple as changing the above line to something like this:

```
default_comment = u'Bodhi is disabling automatic push to stable due to negative karma. The maintainer may manually if they determine that the issue is not severe.'
text = unicode(config.get('disable_automatic_push_to_stable', default_comment))
```

File: production.ini
Comment (bowlofeggs) [inline production.ini]: I suggest a slightly different wording. Perhaps something like:

Bodhi is disabling automatic push to stable due to negative karma. The maintainer may manually if they determine that the issue is not severe.

This is at your option.

File: development.ini.example
Comment (bowlofeggs) [inline development.ini.example]: I suggest a different wording here, also at your option.

**Review by bowlofeggs — CHANGES_REQUESTED** [review]
The only thing I think we need to change is to make the new setting optional, so that bodhi can keep working without it.

File: production.ini
Comment (trishnaguha) [reply production.ini]: Either of them works for me.

File: bodhi/tests/server/services/test_updates.py
Comment (bowlofeggs) [inline bodhi/tests/server/services/test_updates.py]: One last thing to fix and I think we'll be good to go:

This test doesn't pass if ```development.ini``` doesn't have the new ```disable_automatic_push_to_stable``` setting. An easy way to fix it would be to put the default comment here in the call to ```config.get()``` too. It would be nice to allow development checkouts with existing development.ini files to keep working ☺

In the long run, I'd like to adjust ```bodhi.server.config``` so that all settings have sane defaults in a dictionary that is then overwritten by whatever is found in the config files. This way, all settings always have a default and we can always use ```config.get``` with one argument and can guarantee that it will be defined, regardless of what is in the ini files.

**Review by bowlofeggs — CHANGES_REQUESTED** [review]

File: bodhi/tests/server/services/test_updates.py
Comment (trishnaguha) [reply bodhi/tests/server/services/test_updates.py]: Pushed :-).

**Review by bowlofeggs — APPROVED** [review]
Looks great, thanks @trishnaguha!

**Comment by bowlofeggs** [comment]
After merging this I noticed a typo in the message, so I made a follow up PR:

https://github.com/fedora-infra/bodhi/pull/1316

Also, rather than fixing the typo in four places, I made it so I only had to fix it in three ☺

**Pull request merged**
